pkg/pool/operations/create.go

```diff
@@ -103,5 +105,16 @@ func (oc *OperationsConfig) createPool(cspi *cstor.CStorPoolInstance, r cstor.Ra
 	if err != nil {
 		return errors.Errorf("Failed to create pool.. %s .. %s", string(ret), err.Error())
 	}
+
+	// Set compression at the pool level
+	compressionType := cspi.Spec.PoolConfig.Compression
```

Comment:
Sorry for not giving this point before raising a PR. I verified quickly we can provide compression property during pool creation itself.
```go
ret, err := zfs.NewPoolCreate().
		WithType(ptype).
		WithProperty("cachefile", types.CStorPoolBasePath+types.CacheFileName).
                WithProperty("compression", cspi.Spec.PoolConfig.Compression).
		WithPool(PoolName()).
		WithVdevList(vdevlist).
		Execute()
```
But the value of `cspi.Spec.PoolConfig.Compression` shouldn't be empty.

Reply:
Will do that.